protocols/gossipsub/src/behaviour.rs

```diff
@@ -1425,7 +1425,8 @@ where
                                 // check the flood cutoff
                                 let flood_cutoff = (backoff_time
                                     + self.config.graft_flood_threshold())
-                                    - self.config.prune_backoff();
+                                .checked_sub(self.config.prune_backoff())
+                                .unwrap();
```

Comment:
this seems to be [`unchecked_duration_subtraction`](https://rust-lang.github.io/rust-clippy/master/index.html#unchecked_duration_subtraction) right?  
If so, we are switching a possible unintentional panic for an `unwrap`. I'd give it more context with `expect`.  
(the same applies for the other `.checked_sub().unwrap()` substitutions)

Reply:
Yes I think you are correct. I must have overlooked it (I tried to vet the automatic fixes but you know...)

This could be a bug. I'll look into filing an issue.

Reply:
yeah, makes sense. Thanks Hannes.